src/ht.c

```diff
@@ -15,11 +17,30 @@ const int ht_primes_sizes[] = {
 	389357, 467237, 560689, 672827, 807403, 968897, 1162687,
 	1395263, 1674319, 2009191, 2411033, 2893249, 3471899,
 	4166287, 4999559, 5999471, 7199369
-#else
-	1024,
-#endif
 };
 
+#define HASHFN(ht, k) ((ht)->hashfn ? (ht)->hashfn (k) : (ut32)(ut64)(k))
+#define BUCKET(ht, k) ((HASHFN (ht, k) % (ht)->size))
+#define DUPKEY(ht, k) ((ht)->dupkey ? (ht)->dupkey (k) : (void *)k)
+#define DUPVAL(ht, v) ((ht)->dupvalue ? (ht)->dupvalue (v) : v)
+#define CALCSIZEK(ht, k) ((ht)->calcsizeK ? (ht)->calcsizeK (k) : 0)
+#define CALCSIZEV(ht, v) ((ht)->calcsizeV ? (ht)->calcsizeV (v) : 0)
+#define FREEFN(ht, kv) do { if ((ht)->freefn) { (ht)->freefn (kv); } } while (0)
+// when possible, use the precomputed prime numbers which help with collisions,
+// otherwise, at least make the number odd with |1
+#define COMPUTE_SIZE(idx, sz) ((idx) != UT32_MAX ? ht_primes_sizes[idx] : ((sz) | 1))
```

Comment:
sz|1 ?

Reply:
computersize can be a separate function

Reply:
Ok. The `|1` is explained in the comment.
```
// when possible, use the precomputed prime numbers which help with collisions,
// otherwise, at least make the number odd with |1
```